app/controllers/admin/listings_controller.rb

```diff
@@ -22,9 +21,14 @@ def edit
     def update
       @listing = Listing.find(params[:id])
       handle_publish_status if listing_params[:published]
-      bump_listing(@listing.cost) if listing_params[:action] == "bump"
-      update_listing_details
-      clear_listings_cache
+
+      if listing_params[:action] == "bump"
+        bump_success = Listings::Bump.call(@listing, user: current_user)
+        process_no_credit_left && return unless bump_success
+      end
+
+      @listing.update(listing_params.compact)
```

Comment:
Both of these are methods that moved from `ListingsToolkit` wrappers directly into `Listing` instance methods.